diff --git a/dist/index.d.ts b/dist/index.d.ts
index fe8f6b33586f63aff086bd1593a577087a5e46ae..8520550efe3e31886bd1c047b6566a45c66833a5 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -350,6 +350,7 @@ type DtsConfig = {
      * This option takes higher priority than `compilerOptions` in tsconfig.json
      */
     compilerOptions?: any;
+    respectExternal?: boolean;
 };
 type BannerOrFooter = {
     js?: string;
diff --git a/dist/rollup.js b/dist/rollup.js
index 104c1205f7748ac298fb28beb56f32b37ab8a74d..fc848ad0476f4046bc980e51d3adfb3445af27a4 100644
--- a/dist/rollup.js
+++ b/dist/rollup.js
@@ -7689,6 +7689,7 @@ var getRollupConfig = async (options) => {
         json(),
         ignoreFiles,
         dtsPlugin.default({
+          respectExternal: dtsOptions.respectExternal,
           tsconfig: options.tsconfig,
           compilerOptions: {
             ...compilerOptions,
@@ -7730,7 +7731,7 @@ async function runRollup(options) {
     const bundle = await rollup(options.inputConfig);
     const result = await bundle.write(options.outputConfig);
     logger.success("dts", `\u26A1\uFE0F Build success in ${getDuration()}`);
-    _chunkJL7Z4YNEjs.reportSize.call(void 0, 
+    _chunkJL7Z4YNEjs.reportSize.call(void 0,
       logger,
       "dts",
       result.output.reduce((res, info) => {